Update Badger tests

I would like to use Tove to test Badger's latest version v1.5.0.

From what I'm seeing in the [`util`](https://github.com/dgraph-io/tove/blob/a9db6103db075e7be356560eeea38f9c0c8ee9d0/badger/util/util.go) package and the errors from running the [`run.sh`](https://github.com/dgraph-io/tove/blob/a9db6103db075e7be356560eeea38f9c0c8ee9d0/badger/run.sh) script this test should be updated to use Badger's transaction API and also the recently modified GC API.

```
# github.com/dgraph-io/tove/badger/util
util/util.go:12:21: undefined: badger.KV
util/util.go:25:5: opt.ValueGCRunInterval undefined (type *badger.Options has no field or method ValueGCRunInterval)
util/util.go:26:5: opt.ValueGCThreshold undefined (type *badger.Options has no field or method ValueGCThreshold)
util/util.go:30:13: undefined: badger.NewKV
util/util.go:46:17: undefined: badger.KV
util/util.go:52:18: undefined: badger.KV
util/util.go:52:40: undefined: badger.KVItem
util/util.go:53:14: undefined: badger.KVItem
util/util.go:64:22: undefined: badger.KV
```

@manishrjain @peterstace Any other modifications I should consider?

--- a/badger/checker/checker.go
+++ b/badger/checker/checker.go
@@ -17,13 +17,13 @@ func main() {
 	stdout := strings.Split(string(buf), "\n")
 
 	//checkAtomicUpdateConstency(stdout)
-	//checkBadgerConsistency(stdout)
-	checkBadgerBigWorkloadConsistency(stdout)
+	checkBadgerConsistency(stdout)
+	//checkBadgerBigWorkloadConsistency(stdout)
 }
 
 func checkBadgerBigWorkloadConsistency(stdout []string) {
-	kv := StartBadger()
-	defer func() { Must(kv.Close()) }()
+	db := StartBadger()
+	defer func() { Must(db.Close()) }()
 
 	if len(stdout) == 0 {
 		return
@@ -42,12 +42,12 @@ func checkBadgerBigWorkloadConsistency(stdout []string) {
 	case "start:big":
 		for i := 0; i < KeyCount; i++ {
 			k := ConstructKey(uint16(i))
-			if Exists(kv, k) {
+			if Exists(db, k) {
 				hasValidValue := false
 				for j := 0; j < Versions; j++ {
-					kvItem := MustGet(kv, k)
+					value := MustGet(db, k)
 					v := ConstructValue(uint16(i), uint16(j))
-					if reflect.DeepEqual(kvItem.Value(), v) {
+					if reflect.DeepEqual(value, v) {
 						hasValidValue = true
 						break
 					}
@@ -59,7 +59,7 @@ func checkBadgerBigWorkloadConsistency(stdout []string) {
 		for i := 0; i < KeyCount; i++ {
 			const j = Versions - 1
 			Assert(KeyHasValue(
-				kv,
+				db,
 				ConstructKey(uint16(i)),
 				ConstructValue(uint16(i), uint16(j)),
 			))
@@ -75,8 +75,8 @@ func checkBadgerConsistency(stdout []string) {
 		v2 = []byte("value2")
 	)
 
-	kv := StartBadger()
-	defer func() { Must(kv.Close()) }()
+	db := StartBadger()
+	defer func() { Must(db.Close()) }()
 
 	if len(stdout) == 0 {
 		return
@@ -93,28 +93,28 @@ func checkBadgerConsistency(stdout []string) {
 
 	switch lastMsg {
 	case "start:set-key":
-		if Exists(kv, k1) {
-			Assert(KeyHasValue(kv, k1, v1))
+		if Exists(db, k1) {
+			Assert(KeyHasValue(db, k1, v1))
 		}
 	case "stop:set-key":
-		Assert(KeyHasValue(kv, k1, v1))
+		Assert(KeyHasValue(db, k1, v1))
 	case "start:update-key":
-		Assert(Exists(kv, k1))
-		Assert(KeyHasValue(kv, k1, v1) || KeyHasValue(kv, k1, v2))
+		Assert(Exists(db, k1))
+		Assert(KeyHasValue(db, k1, v1) || KeyHasValue(db, k1, v2))
 	case "stop:update-key":
-		Assert(KeyHasValue(kv, k1, v2))
+		Assert(KeyHasValue(db, k1, v2))
 	case "start:del-key":
-		if Exists(kv, k1) {
-			Assert(KeyHasValue(kv, k1, v2))
+		if Exists(db, k1) {
+			Assert(KeyHasValue(db, k1, v2))
 		}
 	case "stop:del-key":
-		Assert(!Exists(kv, k1))
+		Assert(!Exists(db, k1))
 	case "start:ins-upd-del":
-		if Exists(kv, k1) {
-			Assert(KeyHasValue(kv, k1, v1) || KeyHasValue(kv, k1, v2))
+		if Exists(db, k1) {
+			Assert(KeyHasValue(db, k1, v1) || KeyHasValue(db, k1, v2))
 		}
 	case "stop:ins-upd-del":
-		Assert(!Exists(kv, k1))
+		Assert(!Exists(db, k1))
 	default:
 		Assert(false)
 	}

--- a/badger/util/util.go
+++ b/badger/util/util.go
@@ -4,32 +4,24 @@ import (
 	"encoding/binary"
 	"log"
 	"reflect"
-	"time"
-
 	"github.com/dgraph-io/badger"
 )
 
-func StartBadger() *badger.KV {
-	opt := &badger.DefaultOptions
-	opt.MaxTableSize = 512
+func StartBadger() *badger.DB {
+	opt := badger.DefaultOptions
+	opt.MaxTableSize = 2 << 10
 	opt.NumLevelZeroTables = 1
 	opt.NumLevelZeroTablesStall = 2
 	opt.NumMemtables = 1
 	opt.LevelOneSize = 2 << 20
 	opt.ValueLogFileSize = 5 << 20
-	opt.SyncWrites = false
-
-	// Make GC likely to happen. There is 10MB lower limit on what must be
-	// GC'd, so to get a GC, you'll have to remove this part of the
-	// condition in Badger. See doRunGC in value.go.
-	opt.ValueGCRunInterval = time.Millisecond
-	opt.ValueGCThreshold = 0.001
+	opt.SyncWrites = true
 
 	opt.Dir = "."
 	opt.ValueDir = "."
-	kv, err := badger.NewKV(opt)
+	db, err := badger.Open(opt)
 	Must(err)
-	return kv
+	return db
 }
 
 func Must(err error) {
@@ -43,16 +35,48 @@ func MustBool(b bool, err error) bool {
 	return b
 }
 
-func Exists(kv *badger.KV, k []byte) bool {
-	ok, err := kv.Exists(k)
-	Must(err)
-	return ok
+func Exists(db *badger.DB, k []byte) bool {
+	err := db.View(func(txn *badger.Txn) error {
+		_, err := txn.Get(k)
+		return err
+	})
+
+	if err != nil && err != badger.ErrKeyNotFound {
+		log.Fatal(err)
+	}
+
+	return err == nil
 }
 
-func MustGet(kv *badger.KV, k []byte) *badger.KVItem {
-	item := new(badger.KVItem)
-	Must(kv.Get(k, item))
-	return item
+func MustGet(db *badger.DB, k []byte) []byte {
+	var value []byte
+	Must(db.View(func(txn *badger.Txn) error {
+		item, err := txn.Get(k)
+		if err != nil {
+			return err
+		}
+
+		value, err = item.ValueCopy(value)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}))
+
+	return value
+}
+
+func MustSet(db *badger.DB, k []byte, v []byte) {
+	Must(db.Update(func(txn *badger.Txn) error {
+		return txn.Set([]byte(k), []byte(v))
+	}))
+}
+
+func MustDelete(db *badger.DB, k []byte) {
+	Must(db.Update(func(txn *badger.Txn) error {
+		return txn.Delete(k)
+	}))
 }
 
 func Assert(b bool) {
@@ -61,11 +85,10 @@ func Assert(b bool) {
 	}
 }
 
-func KeyHasValue(kv *badger.KV, k, v []byte) bool {
-	Assert(Exists(kv, k))
-	item := MustGet(kv, k)
-	Assert(reflect.DeepEqual(k, item.Key()))
-	return reflect.DeepEqual(v, item.Value())
+func KeyHasValue(db *badger.DB, k, v []byte) bool {
+	Assert(Exists(db, k))
+	value := MustGet(db, k)
+	return reflect.DeepEqual(v, value)
 }
 
 var (

--- a/badger/workload/workload.go
+++ b/badger/workload/workload.go
@@ -13,22 +13,22 @@ func main() {
 
 	Must(os.Chdir("workload_dir"))
 	//atomicUpdateWorkload()
-	//badgerWorkload()
-	badgerBigWorkload()
+	badgerWorkload()
+	//badgerBigWorkload()
 }
 
 func badgerBigWorkload() {
 	fmt.Println("start:big")
-	kv := StartBadger()
+	db := StartBadger()
 	for j := 0; j < Versions; j++ {
 		for i := 0; i < KeyCount; i++ {
 			key := ConstructKey(uint16(i))
 			val := ConstructValue(uint16(i), uint16(j))
-			Must(kv.Set(key, val, 0))
+			MustSet(db, key, val)
 		}
 	}
 	time.Sleep(100 * time.Millisecond)
-	Must(kv.Close())
+	Must(db.Close())
 	fmt.Println("stop:big")
 }
 
@@ -41,29 +41,29 @@ func badgerWorkload() {
 	)
 
 	fmt.Println("start:set-key")
-	kv := StartBadger()
-	Must(kv.Set(k1, v1, 0))
-	Must(kv.Close())
+	db := StartBadger()
+	MustSet(db, k1, v1)
+	Must(db.Close())
 	fmt.Println("stop:set-key")
 
 	fmt.Println("start:update-key")
-	kv = StartBadger()
-	Must(kv.Set(k1, v2, 0))
-	Must(kv.Close())
+	db = StartBadger()
+	MustSet(db, k1, v2)
+	Must(db.Close())
 	fmt.Println("stop:update-key")
 
 	fmt.Println("start:del-key")
-	kv = StartBadger()
-	Must(kv.Delete(k1))
-	Must(kv.Close())
+	db = StartBadger()
+	MustDelete(db, k1)
+	Must(db.Close())
 	fmt.Println("stop:del-key")
 
 	fmt.Println("start:ins-upd-del")
-	kv = StartBadger()
-	Must(kv.Set(k1, v1, 0))
-	Must(kv.Set(k1, v2, 0))
-	Must(kv.Delete(k1))
-	Must(kv.Close())
+	db = StartBadger()
+	MustSet(db, k1, v1)
+	MustSet(db, k1, v2)
+	MustDelete(db, k1)
+	Must(db.Close())
 	fmt.Println("stop:ins-upd-del")
 }
 

